### PR TITLE
Rename CreateTextReader to CreateTextLoader

### LINKED ISSUE #1690: Rename mlContext.Data.TextReader() to mlContext.Data.CreateTextLoader()
Since the object being created is a "TextLoader", I'd recommend the method's name to say so, instead of "TextReader" which looks like a different type. 

I mean code like the following feels confusing:

```
            TextLoader textLoader = mlContext.Data.TextReader(new TextLoader.Arguments()
                                                    {
                                                        Separator = "tab",
                                                        HasHeader = true,
                                                        Column = new[]
                                                                    {
                                                                    new TextLoader.Column("Label", DataKind.Bool, 0),
                                                                    new TextLoader.Column("Text", DataKind.Text, 1)
                                                                    }
                                                    });
```

In addition to that, methods should have a verb as part of the name, so if what it is doing is to create a TextReader, let's say so, as:

mlContext.Data.CreateTextLoader()

We're not being consistent now when having some methods with a verb but other methods just a noun:

image

It is also against C# conventions (and most languages). A method's name should have a verb describing the action being performed by that method:

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods

### PR DESCRIPTION
**Pull request by najeeb-kazmi**
Fixes #1690 

Rename `CreateTextReader` to `CreateTextLoader` to conform to the return type:

```csharp
TextLoader textLoader = mlContext.Data.CreateTextLoader(new TextLoader.Arguments() ...
```

### CODE REVIEW

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs [hunk 1]
```diff
@@ -11,33 +11,33 @@ namespace Microsoft.ML
     public static class TextLoaderSaverCatalog
     {
         /// <summary>
-        /// Create a text reader <see cref="TextLoader"/>.
+        /// Create a text loader <see cref="TextLoader"/>.
         /// </summary>
         /// <param name="catalog">The <see cref="DataOperations"/> catalog.</param>
```
Comment (wschin) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: The doc string to `catalog` is not clear. Could you make a better one? It doesn't describe its relation to other operators and the text loader being created.

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs [hunk 2]
```diff
@@ -11,33 +11,33 @@ namespace Microsoft.ML
...
         /// <param name="columns">The columns of the schema.</param>
```
Comment (wschin) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: `columns` doc string is not very useful too. Can I feed arbitrary columns into this function?

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs [hunk 3]
```diff
@@ -11,33 +11,33 @@ namespace Microsoft.ML
...
         /// <param name="hasHeader">Whether the file has a header.</param>
         /// <param name="separatorChar">The character used as separator between data points in a row. By default the tab character is used as separator.</param>
         /// <param name="dataSample">The optional location of a data sample.</param>
```
Comment (wschin) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: Humm, what is `dataSample`? I know it was not you, but we need to gradually increase the quality of our doc.

Comment (sfilipi) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: >/// <param name="dataSample">The optional location of a data sample.</para [](start = 8, length = 74)

?Do you want to also create a small sample, for the apis of this file, and link it from here?

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs [hunk 4]
```diff
@@ -11,33 +11,33 @@ namespace Microsoft.ML
...
-        public static TextLoader CreateTextReader(this DataOperations catalog,
+        public static TextLoader CreateTextLoader(this DataOperations catalog,
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: >CreateTextLoader [](start = 33, length = 16)

My personal take on this is: TextReader is a much better name than TextLoader. I would rather rename the  TextLoader class to TextReader.. is that ok for the cmd usage.

Comment (TomFinley) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: This is correct, we are replacing the `IDataLoader` idiom with the `IDataReader` idiom, as part of #581 and much subsequent work.

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs (hunk 1 above)
Comment (najeeb-kazmi) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: This is actually an extension method to MLContext. The doc string for catalog is not visible to the user.

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs (hunk 4 above)
Comment (najeeb-kazmi) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: `TextReader` is conflated with `System.IO.TextReader` and should be avoided. I've filed #2144 for renaming the class. I propose making the method name and return type consistent in this PR and revisiting the name when we rename the class.

cc: @glebuk @eerhardt

Comment (TomFinley) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: Ah. Good point,. Maybe we should actually yname `IDataReader` to `IDataLoader`? And call the existing legacy `IDataLoader` interface something like `IDataLoaderLegacy`? What we all think?

Comment (TomFinley) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: Similar renames to `IDataReaderEstimator` to I suppose `IDataLoaderEstimator`.

Comment (CESARDELATORRE) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs]: My point is to be consistent between method's name and returned type, so either on of the following would be good for me: 👍 

public static Text**Loader** CreateText**Loader**()

public static Text**Reader** CreateText**Reader**()

**Pull request merged**